External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -1861,6 +1861,49 @@ void OpDispatchBuilder::PopcountOp(OpcodeArgs) {
   SetRFLAG<FEXCore::X86State::RFLAG_OF_LOC>(Zero);
 }
 
+void OpDispatchBuilder::XLATOp(OpcodeArgs) {
+  OrderedNode *Src = _LoadContext(8, offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RBX]), GPRClass);
+  OrderedNode *Offset = _LoadContext(1, offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RAX]), GPRClass);
+
+  if (Op->Flags & FEXCore::X86Tables::DecodeFlags::FLAG_FS_PREFIX) {
+    Src = _Add(Src, _LoadContext(8, offsetof(FEXCore::Core::CPUState, fs), GPRClass));
+  }
+  else if (Op->Flags & FEXCore::X86Tables::DecodeFlags::FLAG_GS_PREFIX) {
+    Src = _Add(Src, _LoadContext(8, offsetof(FEXCore::Core::CPUState, gs), GPRClass));
+  }
+  Src = _Add(Src, Offset);
+
+  auto Res = _LoadMem(GPRClass, 1, Src, 1);
+
+  _StoreContext(GPRClass, 1, offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RAX]), Res);
+}
+
+template<bool FS>
```

Comment:
For code readability purposes, could it be possible to use an Enum here?

That way we would end up with something like this below: 

```c++
    {OPD(FEXCore::X86Tables::TYPE_GROUP_15, PF_F3, 0), 1, &OpDispatchBuilder::ReadSegmentReg<Segment_FS)>},
    {OPD(FEXCore::X86Tables::TYPE_GROUP_15, PF_F3, 1), 1, &OpDispatchBuilder::ReadSegmentReg<Segment_GS>},
    {OPD(FEXCore::X86Tables::TYPE_GROUP_15, PF_F3, 2), 1, &OpDispatchBuilder::WriteSegmentReg<Segment_FS>},
    {OPD(FEXCore::X86Tables::TYPE_GROUP_15, PF_F3, 3), 1, &OpDispatchBuilder::WriteSegmentReg<Segment_GS>},
```

Reply:
Concern addressed